### PR TITLE
HTTP Header Link doesn't support rel=icon

### PR DESCRIPTION
**Pull request by James-E-A**
### Description

Note that `rel=icon` is not supported for HTTP Link headers; it's only supported for link elements.

### Motivation

I was bit by this when trying to modify the favicon for **specific** non-HTML documents such as PDF.

### Additional details

Live test / demo URL: http://webpages.uah.edu/~je0029/archive/dubfire.net/ssl-mitm.pdf

### Related issues and pull requests

_(none)_

### CODE REVIEW

**Review by arely222 — COMMENTED** [review]
#24865

**Comment by Elchi3** [comment]
> for specific non-HTML documents such as PDF.

Maybe it only doesn't work for non-HTML things?

I think it does work for HTML and so as it stands this PR would be incorrect.

**Comment by JamesTheAwesomeDude** [comment]
is this a Firefox bug&mdash;is it the browser that should be changed, <em>rather than</em> the documentation?

**Comment by Josh-Cena** [comment]
@JamesTheAwesomeDude The way to confirm this is (1) test on other browsers and check if they have the same behavior (2) search the Firefox issue tracker and see if they confirm it's a bug (in which case maybe we can add BCD data) or if they give an explanation on why their current behavior is intended (in which case we can update the content) (3) send an issue if one doesn't exist and get feedback.

**Comment by jfrech** [comment]
[RFC 8288](https://www.rfc-editor.org/rfc/rfc8288.html#section-4.1) declares [Link relations](https://www.iana.org/assignments/link-relations/link-relations.xhtml) as the registry for HTTP "Link" headers, thereby defining [HTML semantics](https://html.spec.whatwg.org/multipage/links.html#rel-icon) for it.

Morally speaking, linking is intentionally a feature to inter-link different protocols, so an HTML-only link does not make a whole lot of sense.  
Adding a note that common-place browsers do not support rendering icon linkage to non-HTML resources is nonetheless surely a good idea.

**Comment by jfrech** [comment]
> Obviously, it's _irrelevant_ in the case of HTML — since you could just insert the tag in the `head` — however, my edit is correct as-written, as far as I can tell.

@JamesTheAwesomeDude Can you confirm that two different HTML pages with two different HTML `<link rel=icon ...>`-linked favicons are correctly rendered?

**Comment by JamesTheAwesomeDude** [comment]
> Can you confirm that two different HTML pages with two different HTML `<link rel=icon ...>`-linked favicons are correctly rendered?

...yes? That's how web apps like Gmail and Discord change the tab icon to show unread indicators, etc. -- If that were broken, everyone would have submitted a bug report already. But I'll demo it here at your request:

 - http://webpages.uah.edu/~je0029/tmp/example_1.html
 - http://webpages.uah.edu/~je0029/tmp/example_2.html

**Comment by JamesTheAwesomeDude** [comment]
> (1) test on other browsers and check if they have the same behavior

Edge v113 and Chrome v113 behave the same as Firefox v102 here (i.e., respecting the tag form while not respecting the header form).

> (2) search the Firefox issue tracker and see if they confirm it's a bug

It looks like there's nothing there (but IME bugzilla's search is very unreliable so this is a soft-negative https://bugzilla.mozilla.org/buglist.cgi?quicksearch=rel+icon+header)

> (3) send an issue if one doesn't exist and get feedback.

That'll take a while.

For now, the document as-written is incorrect and misleading as regards the behavior of all major browsers. Slight quibbles about completeness aside, is there anything incorrect or misleading about my proposed edit?

**Comment by jfrech** [comment]
> Slight quibbles about completeness aside, is there anything incorrect or misleading about my proposed edit?
Yes; you mix taking about the specification of `Link: ` with its current real-world browser support for `rel=icon`.

**Comment by jfrech** [comment]
@JamesTheAwesomeDude I have just looked at one of my gmail account's logged-in HTML and found its favicon to be static:
```html
<link rel="shortcut icon" href="https://ssl.gstatic.com/ui/v1/icons/mail/rfr/gmail.ico" type="image/x-icon">
```
Furthermore, the `/favicon.ico` endpoint could be dynamic to implement notification counts.

File: files/en-us/web/http/headers/link/index.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ browser-compat: http.headers.Link
 
 {{HTTPSidebar}}
 
-The HTTP **`Link`** entity-header field provides a means for serializing one or more links in HTTP headers. It is semantically equivalent to the HTML {{HTMLElement("link")}} element.
+The HTTP **`Link`** entity-header field provides a means for serializing one or more links in HTTP headers. It is semantically equivalent to the HTML {{HTMLElement("link")}} element, except that `rel=icon` is not supported.
```
Comment (estelle) [inline files/en-us/web/http/headers/link/index.md:10]:
```suggestion
The HTTP **`Link`** entity-header field provides a means for serializing one or more links in HTTP headers. It is semantically equivalent to the HTML {{HTMLElement("link")}} element.

> **Note:** Some browsers do not support the "icon" relation via HTTP ([FF#1185705](https://bugzilla.mozilla.org/show_bug.cgi?id=1185705)) . Use HTML `<link rel="icon">` instead.
```

**Review by teoli2003 — APPROVED** [review]
👍

**Pull request merged**